plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -1330,23 +1341,26 @@ int amdgpu_plugin_dump_file(int fd, int id)
 	if (!buf) {
 		pr_perror("failed to allocate memory\n");
 		ret = -ENOMEM;
-		goto failed;
+		goto exit;
 	}
 
 	criu_kfd__pack(e, buf);
 
 	ret = write_file(img_path,  buf, len);
-	if (ret)
-		ret = -1;
 
 	xfree(buf);
 exit:
-failed:
+	/* Restore all queues */
+	pause_process(fd, false);
+
 	sys_close_drm_render_devices(&src_topology);
 	free_e(e);
-	pr_info("amdgpu_plugin: Exiting from dumper for fd = %d\n", major(st.st_rdev));
-        return ret;
 
+	if (ret)
+		pr_err("amdgpu_plugin: Failed to dump (ret:%d)\n", ret);
```

Comment:
I think write_file will provide a valid error code so lets just use pr_perror for consistency.

Reply:
This is meant to print the error code for any failure in this function. write_file calls pr_perror to print the errno